### ISSUE TITLE
Some improvements during work with the database

### ISSUE THREAD

**Issue by xgreenx** [FuelLabs/fuel-core#622]
Use `Option<&[u8]>` instead of `Option<Vec<u8>>` in the `iter_all` of `fuel_core::state::KeyValueStore`. Also decide, do we really need usage of `Option`? If `len` is zero it is the same as `None`. Apply the same change for all upper functions.

<img width="759" alt="image" src="https://user-images.githubusercontent.com/18346821/190230186-4b00ecd8-56cc-4b67-a997-61bda212a0e7.png">

Also there in the implementation of the iterator, we can avoid the copy of the slices into vectors. Also think about [comment](https://github.com/FuelLabs/fuel-core/pull/591#discussion_r970116644).

<img width="793" alt="image" src="https://user-images.githubusercontent.com/18346821/190230122-eed1758c-75f1-4312-9771-2da7a55f28fd.png">

Minor improvement in `Database::insert`:
<img width="904" alt="image" src="https://user-images.githubusercontent.com/18346821/190273055-c90112e0-7388-4a9c-ae82-c22296614fbb.png">

To use the same naming across the project in `Database::exists`:
<img width="903" alt="image" src="https://user-images.githubusercontent.com/18346821/190273090-697510e9-29fb-4a49-bf7f-02bdade96415.png">

**Comment by Voxelot** [comment]
>  If len is zero it is the same as None

Not necessarily, there is value in being able to distinguish between empty values and non-existent ones.

**Comment by xgreenx** [comment]
@Voxelot What do you think about renaming all `Database`'s methods by underscore? To highlight that they are internal methods, and you need to work with `StorageMutate` and `StorageInspect`. 

- `Database::insert` -> `Database::_insert`
- `Database::remove` -> `Database::_remove`
- `Database::get` -> `Database::_get`
- `Database::exists` -> `Database::_contains_key`
- `Database::iter_all` -> `Database::_iter_all`

**Comment by AliJafriETH** [comment]
Hey @xgreenx & @Voxelot, is the issue still open. i'd like to work on it :))

**Comment by xgreenx** [comment]
There is in progress PR https://github.com/FuelLabs/fuel-core/pull/643 that is blocked by refactoring https://github.com/FuelLabs/fuel-core/issues/809. The change is almost ready, so it is better to select another issue=D
